polkadot/node/network/collator-protocol/src/collator_side/mod.rs

```diff
@@ -422,7 +443,22 @@ async fn distribute_collation<Context>(
 		);
 	}
 
-	let our_core = our_cores[0];
+	// Double check that the specified `core_index` is among the ones our para has assignments for.
+	if !our_cores.iter().any(|assigned_core| assigned_core == &core_index) {
```

Comment:
We seem to be fetching core information 3 times: Once in lookahead, then in generation, then here. The only reason to fetch it here is actually to get the core count.

Future refactor: It might make sense to fetch the relevant information once, collect it in a struct and then just pass it around.

Reply:
https://github.com/paritytech/polkadot-sdk/issues/3838